### DISCUSSION TITLE
Using Apple2-IO-RPi with an Apple II Plus

### DISCUSSION THREAD

**Discussion by k6lcm** [tjboldt/Apple2-IO-RPi#142] [General]
I know the docs say that the A2-IO-RPi assumes an Apple IIe with an 80 column card. Has anyone used it with an Apple II Plus with only 40 column support? Does it work at all? I can always try it in my IIGS but I am interested to see if I can expand the capabilities of my II Plus.

**Comment by tjboldt** [comment]
The shell could be modified to better support 40 column mode and shift-key mod on the II Plus but that hasn't been a priority so far. Currently it will try to call $C300 which will not work so well without the 80 column firmware. Also, it would need to add shift-key mod support as trying to use a Linux command prompt in upper case is all but useless. In theory, the drive emulation should work on a II Plus and if you execute commands via RPI.COMMAND, it should also work.

**Reply by k6lcm** [reply]
Thanks for the detailed reply. All makes sense. Would it be feasible and actually easier for the program to interpret all keyboard input from the II+ as lowercase instead?

I'll play around and see what I encounter. But it sounds like learning as much as I can on the IIGS makes the most sense ATM.

**Reply by JPT77** [reply]
For me it works fine. Boots the prodos image or Totalreplay just fine.
You need a character ROM that can display lower case letters to see the boot up message from the Apple2IORPI.
What is not workin is writing RPI commands because of missing lower case input. One of my AppleIIplus should have lower case input?!
In this regard your idea would probably solve the problem.

**Reply by Hubidrei** [reply]
Ok, wit my Apple][ clone the call $C300 lets my computer hang since it has 80column but not in slot #3. 
I can boot into 80col mode then overwrite the JMP $C300 with nop and change the address for 80col. cursor position to get it work.
